AddressSpace/templates/designToSourceVariablesBody.jinja

```diff
@@ -0,0 +1,411 @@
+                device->lock();
+              {% elif sv.get('addressSpaceWriteUseMutex') == 'of_parent_of_containing_object' %}
+                device->getParent()->lock();
+              {% endif %}
+              
+              /* OK - Device Logic linked. */
+              try
+              {
+                s = device->write{{sv.get('name')|capFirst}} (value );
+              }
+              catch (...)
+              {
+                /* TODO -- how to signalize error from here */
+                LOG(Log::ERR) << "An exception was thrown from write{{sv.get('name')|capFirst}}";
+                s = OpcUa_BadInternalError;
+              }
+              {% if sv.get('addressSpaceWriteUseMutex') == 'of_this_operation' %}
+                device->unlockVariableWrite_{{sv.get('name')}} ();
+              {% elif sv.get('addressSpaceWriteUseMutex') == 'of_this_variable' %}
+                device->unlockVariable_{{sv.get('name')}} ();
+              {% elif sv.get('addressSpaceWriteUseMutex') == 'of_containing_object' %}
+                device->unlock();
+              {% elif sv.get('addressSpaceWriteUseMutex') == 'of_parent_of_containing_object' %}
+                device->getParent()->unlock();
+              {% endif %}
+            }
+            else
+              s = OpcUa_BadInternalError; // device link was null.
+          }
+          else
+            s = OpcUa_BadInternalError; // invalid dynamic_cast, probably internal quasar error...
+          }
+        else
+          s = OpcUa_BadDataEncodingInvalid; // conversion from variant impossible.
+        UaDataValue result (UaVariant(value), s.statusCode(), UaDateTime::now(), UaDateTime::now());
+        // get appropriate object
+        s = m_callback->finishWrite (
+          m_hTransaction,
+          m_callbackHandle,
+          s
+        );
+        LOG(Log::TRC) << "After finishWrite status:" << s.toString().toUtf8() << endl;
+        
+        }
+        
+        virtual std::string describe() const
+        {
+          return std::string("write sourcevariable {{sv.get('name')}} of object ") + m_parentObjectNode->nodeId().toString().toUtf8();
+        }
+        
+        
+      private:
+        IOManagerCallback* m_callback;
+        OpcUa_UInt32       m_hTransaction;
+        OpcUa_UInt32       m_callbackHandle;
+        const UaNode*      m_parentObjectNode;
+        UaVariant          m_variant;
+    };
+  {% endfor %}
+  
+{% endfor %}
+
+UaStatus SourceVariables_spawnIoJobRead (		
+  ASSourceVariableJobId jobId,
+  IOManagerCallback *callback,
+  OpcUa_UInt32 hTransaction,
+  OpcUa_UInt32        callbackHandle,
+  const UaNode *parentNode
+)
+{
+  if (! sourceVariableThreads)
+    throw std::runtime_error("Attempted Source Variable operation, but Source Variable threads are not up.");
+  switch (jobId)
+  {
+    {% for className in designInspector.get_names_of_all_classes() %}
```

Comment:
another macro possible here? This logic (with all the jinja ifs and fors) looks pretty similar to the logic in the method below.